### ISSUE TITLE
Incompatible packages on Windows

### ISSUE THREAD

**Issue by synapticarbors** [conda-forge/magic-wormhole-feedstock#6]
Attempting to create an environment with `magic-wormhole` on Windows, I get a pretty unhelpful list of incompatibility errors:

```
$ conda create -n wormhole magic-wormhole --override-channels -c conda-forge
Collecting package metadata (current_repodata.json): done
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: \
Found conflicts! Looking for incompatible packages.
This can take several minutes.  Press CTRL-C to abort.
failed

UnsatisfiableError: The following specifications were found to be incompatible with each other:

Package tqdm conflicts for:
magic-wormhole -> tqdm[version='>=4.13.0']
Package autobahn conflicts for:
magic-wormhole -> autobahn[version='>=0.14.1']
Package pynacl conflicts for:
magic-wormhole -> pynacl
Package python conflicts for:
magic-wormhole -> python
Package service_identity conflicts for:
magic-wormhole -> service_identity
Package six conflicts for:
magic-wormhole -> six
Package twisted conflicts for:
magic-wormhole -> twisted[version='>=17.5.0']
Package automat conflicts for:
magic-wormhole -> automat
Package humanize conflicts for:
magic-wormhole -> humanize
Package attrs conflicts for:
magic-wormhole -> attrs[version='>=16.3.0']
Package spake2 conflicts for:
magic-wormhole -> spake2==0.8
Package click conflicts for:
magic-wormhole -> click
Package hkdf conflicts for:
magic-wormhole -> hkdf
```

I've looked through all of these dependencies, and I don't see that any of them are missing (using `conda search`). I tried specifying both `python=3.6` and `python=2.7`, but got a similar message. This environment is created just fine using the same command on MacOS.

**Comment by CJ-Wright** [comment]
I think the issue here is to get libsodium working on windows properly.

**Comment by synapticarbors** [comment]
@CJ-Wright can you tell that from the error message and list of conflicts?

**Comment by CJ-Wright** [comment]
No, but I remember windows being an issue

**Comment by CJ-Wright** [comment]
Or maybe it was one of the other deps. I remember that getting the deps on windows was a pain.

**Comment by CJ-Wright** [comment]
I have no idea what is going on with spake here https://github.com/conda-forge/spake2-feedstock

**Comment by synapticarbors** [comment]
There are files available for spake2 on anaconda.org though:
https://anaconda.org/conda-forge/spake2/files

**Comment by CJ-Wright** [comment]
I think it is coming from python-spake2

**Comment by synapticarbors** [comment]
Yeah, that seems correct. Is there a problem with `python` not being a `run` requirement there?

https://github.com/conda-forge/python-spake2-feedstock/blob/master/recipe/meta.yaml

**Comment by CJ-Wright** [comment]
that could do it

**Comment by CJ-Wright** [comment]
Would you be interested in putting in a PR fixing it?

**Comment by synapticarbors** [comment]
I'll give it a try

**Comment by CJ-Wright** [comment]
you might consider giving [test downstreams](https://docs.conda.io/projects/conda-build/en/latest/resources/define-metadata.html#downstream-tests) a try

**Comment by synapticarbors** [comment]
Unfortunately the change to `spake8` didn't do anything. I get the same errors even with the new binary posted.

**Comment by CJ-Wright** [comment]
Shoot. One thing to try is to put up a PR that removes the noarch and then use the CI to play around with things.

**Comment by synapticarbors** [comment]
After looking at the successful build on Windows for py3.6, I realized that the build system was pulling `vc` and `vs2015_runtime` from the `defaults` channel. If I instead run the following on my local machine, I can get it to work:

```
$ conda create -n wormhole magic-wormhole --override-channels -c conda-forge -c defaults
```

Should the `defaults` channel be necessary or does that indicate that something is missing on the conda-forge side that should be there?

**Comment by CJ-Wright** [comment]
hmm, not certain, usually we put conda-forge before defaults with strict, is your conda setup like that?

**Comment by CJ-Wright** [comment]
(also you should consider adding yourself as maintainer to these packages)

**Comment by synapticarbors** [comment]
I tend to have `defaults` and then `conda-forge`, but only use `conda-forge` to pick up things that are pure-python, unless I'm doing something specific with an isolated environment. Often times then I'll use `--override-channels` and just pick the specific channel I want. 

That said, I see `vc` on anaconda.org under conda-forge, although the latest build is labeled `broken`. That's also the case for `vs2015_runtime`. There are non-broken files, but they only have `cf201901` labels and not `main`. This might be the problem of why it has to go to `defaults` but I'm not sure.

**Comment by hmaarrfk** [comment]
in 2025, i believe that this is no longer an issue. The constraints we had in 2020 are no longer here.

Please lets open a new issue if this persists.
